### PR TITLE
log the profiler config at agent start.

### PR DESCRIPTION
**Pull request by breedx-splk**
This helps us to do troubleshooting and provide additional support.

### CODE REVIEW

File: profiler/src/test/java/com/splunk/opentelemetry/profiler/ConfigurationLoggerTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        when(config.getBoolean(CONFIG_KEY_TLAB_ENABLED)).thenReturn(false);
+        when(config.getString(CONFIG_KEY_PERIOD_PREFIX + ".jdk.threaddump")).thenReturn("500ms");
+
+        ConfigurationLogger configurationLogger = new ConfigurationLogger(log);
+
+        configurationLogger.log(config);
+
+        InOrder inOrder = inOrder(log);
+        inOrder.verify(log).info("-----------------------");
+        inOrder.verify(log).info("Profiler configuration:");
```
Comment (jkwatson) [inline profiler/src/test/java/com/splunk/opentelemetry/profiler/ConfigurationLoggerTest.java:48]: any thoughts to using logunit here, rather than mocking?

File: profiler/src/test/java/com/splunk/opentelemetry/profiler/ConfigurationTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package com.splunk.opentelemetry.profiler;
+
+import io.opentelemetry.instrumentation.api.config.Config;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class ConfigurationTest {
```
Comment (jkwatson) [inline profiler/src/test/java/com/splunk/opentelemetry/profiler/ConfigurationTest.java:10]: this change seems unrelated to the logging of the config.

File: profiler/src/test/java/com/splunk/opentelemetry/profiler/ConfigurationLoggerTest.java (hunk 1 above)
Comment (breedx-splk) [reply profiler/src/test/java/com/splunk/opentelemetry/profiler/ConfigurationLoggerTest.java]: yeah that worked out, thanks for the advice.

File: profiler/src/test/java/com/splunk/opentelemetry/profiler/ConfigurationTest.java (hunk 2 above)
Comment (breedx-splk) [reply profiler/src/test/java/com/splunk/opentelemetry/profiler/ConfigurationTest.java:26]: It is. I had factored it out thinking that I would use it in the logging, but it was misleading in the logs, so I didn't end up using it there...but I felt like it was still a good change so I left it. Do you think it should be removed?

Comment (jkwatson) [reply profiler/src/test/java/com/splunk/opentelemetry/profiler/ConfigurationTest.java:26]: doesn't matter to me too much. I was just confused to see this here when it didn't appear to have anything to do with the rest of the PR. :)

**Review by jkwatson — APPROVED** [review]

File: profiler/src/main/java/com/splunk/opentelemetry/profiler/ConfigurationLogger.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** This class logs the active profiler configuration for debug/troubleshooting purposes. */
+public class ConfigurationLogger {
+
+  private static final Logger logger = LoggerFactory.getLogger(ConfigurationLogger.class);
+
+  public void log(Config config) {
+    logger.info("-----------------------");
```
Comment (mateuszrzeszutek) [inline profiler/src/main/java/com/splunk/opentelemetry/profiler/ConfigurationLogger.java:39]: Just one thought: do we want to log everything to `INFO` if it's for debug/troubleshooting purposes?

Comment (mateuszrzeszutek) [reply profiler/src/main/java/com/splunk/opentelemetry/profiler/ConfigurationLogger.java:39]: Or maybe we could guar this with `config.isAgentDebugEnabled()`

Comment (breedx-splk) [reply profiler/src/main/java/com/splunk/opentelemetry/profiler/ConfigurationLogger.java:39]: I did intend for it to be at INFO -- because it's only at app startup and it's pretty important when troubleshooting.
